src/tactic/scc.lean

```diff
@@ -93,47 +201,80 @@ parameter g : expr_map (list $ expr × expr)
 parameter visit : ref $ expr_map bool
 parameter cl : closure
 
+/-- `merge_path path e`, where `path` and `e` forms a cycle with proofs of implication between
```

Comment:
Could you clarify this? I don't really understand. What's the end result of this function? It's `tactic unit`, so it's updating the tactic state in some way.

Reply:
It is actually updating `cl` which is a local `parameter` of type `closure` (i.e. a mutable data structure). The path describes a cycle of the shape:

```
v₀ -> v₁ -> v₂ -> v₃ -> v₀
```

represented as:

```lean
[(v₀, h₀ : v₀ → v₁), (v₁, h₁ : v₁ → v₂), (v₂, h₂ : v₂ → v₃), (v₃, h₃ : v₃ → v₀)], v₀
```

In order to merge the equivalence class of those vertices, I need a two by two proof of equivalence so I start by iteratively applying transitivity so that, for each `v` (including `v₀`) I have a proof both of `v → v₀` and `v₀ → v`. Then I can apply `iff.intro` and have a proof of `v₀ ↔ v` and then I can merge all the equivalence classes with that of `v₀`.

Reply:
I see, thanks.